lib/project/pro_motion/data_table.rb

```diff
@@ -0,0 +1,187 @@
+module ProMotion
+  module DataTable
+    include ProMotion::Styling
+    include ProMotion::Table::Utils
+
+    def table_view
+      self.view
+    end
+
+    def screen_setup
+      set_up_reload_notification
+      set_up_fetch_controller
+
+      # TODO - implement dynamic row height
+      # set_up_row_height
+    end
+
+    def set_up_reload_notification
+      NSNotificationCenter.defaultCenter.addObserver(self,
+        selector: "update_table_data:",
+        name: NSManagedObjectContextDidSaveNotification,
+        object: nil)
+    end
+
+    def set_up_fetch_controller
+      error_ptr = Pointer.new(:object)
+      fetch_controller.delegate = self
+
+      unless fetch_controller.performFetch(error_ptr)
+        raise "Error performing fetch: #{error_ptr[2].description}"
+      end
+    end
+
+    def update_table_data(notification = nil)
+      data_model.send(:update) if data_model.respond_to?(:update)
+      Dispatch::Queue.main.async do
+        fetch_controller.managedObjectContext.mergeChangesFromContextDidSaveNotification(notification)
+      end
```

Comment:
I was following on some of the work of @willrax. I'm assuming that it's so that the fetch_controllers' moc stays in sync with the notification's moc if they're different contexts?